src/main/frontend/handler/editor.cljs

```diff
@@ -2583,9 +2602,18 @@
       nil
 
       :else
-      (do
-        (delete-block-aux! next-block false)
-        (state/set-edit-content! input-id (str value "" (:block/content next-block)))
+      (let [edit-block (state/get-edit-block)
+            transact-opts (cond->
+                            {:outliner-op :delete-block
+                             :concat-data {:last-edit-block (:block/uuid edit-block)
+                                           :end? true}})
+            new-content (str value "" (:block/content next-block))
+            repo (state/get-current-repo)]
+        (outliner-tx/transact! transact-opts
```

Comment:
We can group multiple operations in the same db transaction with `outliner-tx/transact!`.

Reply:
`transact!` could be nested too.